vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,477 @@
+# object, the `output_type_mean_quantile`
+output_type_mean_quantile <- create_output_type(mean_example, quantile_example)
+```
+
+### Creating the `model_task` objects
+
+For both tasks, we want to require a "mean" output, but we want a "quantile"
+output type for the `inc death` target and an optional "median" for the `inc
+hosp` target.
+
+```{r create-model-task}
+#Create an object of class `model_task` representing a model task, the `model_task_hosp`
+model_task_hosp <- create_model_task(
+  task_ids = task_ids_example,
+  output_type = output_type_mean_median,
+  target_metadata = target_metadata_example
+)
+
+#Create another object of class `model_task` representing a similar model task, the `model_task_death`
+model_task_death <- create_model_task(
+  task_ids = task_ids_example,
+  output_type = output_type_mean_quantile,
+  target_metadata = target_metadata_example
+)
+
+# Combine the `model_task` objects to create a `model_tasks` class object, the
+# `model_task_example`
+model_task_example <- create_model_tasks(model_task_hosp, model_task_death)
+```
+Now that we have a set of model tasks, we can create rounds for them.
+
+### Creating the `round` objects
+
+While the model tasks define what a model output submission should look like,
+the round defines the cadence of submissions and the timeframe under which a
+submission is considered valid.
+
+The most common way to create rounds is to create a single `round` object that
+tracks a date variable, in this case `"origin_date"`. The submission window for a given round is a period of one week, where the beginning of the window is
+4 days before the origin date and the end of the submission window is 2 days
+after the origin date. This means if you have an origin date of 
+`r format(as.Date("2023-01-02"), "%A %Y-%m-%d")`, then valid submission dates
+for that round are from 
+`r paste(format(as.Date("2023-01-02") + c(-4, 2), "%A %Y-%m-%d"), collapse = " to ")`
+
+```{r create-rounds}
+# Create a representation of a round as a list object of class `round`, the
+# `round1`
+round1 <- create_round(
+  round_id_from_variable = TRUE,
+  round_id = "origin_date",
+  round_name = "Round 1",
+  model_tasks = model_task_example,
+  submissions_due = list(
+    relative_to = "origin_date",
+    start = -4L,
+    end = 2L
+  )
+)
+
+# Combine the `round` objects to create a `rounds` class object, the `rounds`
+rounds <- create_rounds(round1)
+```
+
+### Creating and saving the `tasks` config file
+
+And the penultimate step is to create a config file and write it out to your
+hub.
+
+```{r create-tasks, eval = FALSE}
+# Create a representation of a complete `"tasks"` config file as a list object
+# of class `config`
+config <- create_config(rounds, output_type_id_datatype = "auto")
+
+# Write the `config` object to a JSON file (note that this will overwrite any
+# existing file with the same name in the "hub-config" folder)
+write_config(config, overwrite = TRUE)
```

Comment:
```suggestion
write_config(config)
```
Not necessary the first time a file is written. Might in fact be useful to demonstrate that the default settings protect you from overwriting and that `overwrite` argument allows you to overwrite

Reply:
good point! Given that this is generating a file from scratch, I'll leave `overwrite` out.